In [1]:
import pandas as pd
import math
import git
import sys
import statistics
repo = git.Repo("./", search_parent_directories=True)
homedir = repo.working_dir
data = pd.read_csv('modified_submission2.csv')
data.to_csv('modified_submission3.csv', index=False)

In [2]:
df = pd.read_csv('modified_submission3.csv').set_index('id').sort_index()
sample_submission = pd.read_csv(f"{homedir}/sample_submission.csv").set_index('id').sort_index()

In [3]:
df.isnull().values.any()

False

In [4]:
print(df.index.equals(sample_submission.index))

True


In [5]:
df1 = pd.read_csv('formatted_submission.csv').set_index('id').sort_index()
df1.isnull().values.any()
print(df1.index.equals(sample_submission.index))

True


In [6]:
def get_date(str1):
    counter = 0
    month = ""
    day = ""
    for x in str1:
        if counter == 1:
            month += x
        elif counter == 2:
            day += x
        if x == '-':
            counter += 1
    month = month[:-1]
    day = day[:-1]
    return (month, day)  

def get_fips(code):
    counter = 0
    fips = ""
    for x in code:
        if counter == 3:
            fips += x
        if x == '-':
            counter += 1
    return fips

In [7]:
for row in df1.iterrows():
    row0 = row[0]
    row = row[1]
    for i in range(0, 9):
        if row[i] > 1000000:
            print(row0)

In [8]:
counter = 0
for row in df1.iterrows():
    row0 = row[0]
    row = row[1]
    
    date = get_date(row0)
    if int(date[0]) == 5:
        if int(date[1]) > 10:
            break
        
    elif int(date[0]) > 5:
        break
    
    badRow = False
    if row[-1] > 5000:
        badRow = True
    if badRow == True:
        counter += 1
        print(row0)
    """
    if row[0] == 0 and row[8] > 1000:
        print(row0)
    """

2020-04-09-36061
2020-04-10-36061
2020-04-11-36061
2020-04-12-36061
2020-04-13-36061
2020-04-14-36061
2020-04-15-36061
2020-04-16-36061
2020-04-17-36061
2020-04-18-36061
2020-04-19-36061
2020-04-20-36061
2020-04-21-36061
2020-04-22-36061
2020-04-23-36061
2020-04-24-36061
2020-04-25-36061
2020-04-26-36061
2020-04-27-36061
2020-04-28-36061
2020-04-29-36061
2020-04-30-36061
2020-05-01-36061
2020-05-02-36061
2020-05-03-36061
2020-05-04-36061
2020-05-05-36061
2020-05-06-36061
2020-05-07-36061
2020-05-08-36061
2020-05-09-36061
2020-05-10-36061
2020-05-10-42011


The counties with predicted deaths > 10000 at some point in the next two weeks are
1. 36061, this is alright since it is new york
2. 21207, russell county kentucky which has 0 deaths so far and less than 20 cases...makes no sense
3. 1037, coosa county alabama which is rural having 25 cases and 1 death...makes no sense
4. 48107,crosby county texas which has 1 case and 1 death ... makes no sense
3. 18031. decatur county which is pretty bad, 194 cases and 26 deaths, I could believe it. Dividing by 100
So we set the values to "3" for 21207, "3" for 1037 and "3" for 48107
Additional counties with predicted deaths > 5000 at some point in the next two weeks are
1. 42055. franklin county pa which has 152 cases and 10 deaths, I could believe it. Dividng by 100
2. 47159. smith county tennessee which has 19 cases and 1 death...setting values to "3"


Counties with largest prediction > 5000:
1. 5061: hempstead county -> 2-3 category
2. 21207: russell county kentucky -> 2-3 category
3. 39075: holmes county oh -> 2-3 category
4. 48365: panola county texas -> 5-10 category
5. 42011: berks county pa. This is legit

In [9]:
print(counter)

33


In [10]:
from scipy.stats import norm
from scipy import stats
mean, sigma = 3.0, 0.5

conf_int = stats.norm.interval(0.0, loc=mean, 
    scale=sigma)

print(conf_int[0])

3.0


In [11]:
norm.ppf()

TypeError: ppf() missing 1 required positional argument: 'q'

In [ ]:
min(34, 5)